docs/spec_draft.md

```diff
@@ -703,6 +704,76 @@ matrix, then the behavior is undefined.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.compare
+
+### Semantics
+
+Performs element-wise comparison of `lhs` and `rhs` tensors according to
+`comparison_direction` and `compare_type`, and produces a `result` tensor.
+
+The values of `comparison_direction` have the following semantics:
+  * `EQ`: `lhs` $=$ `rhs`.
+  * `NE`: `lhs` $\ne$ `rhs`.
+  * `GE`: `lhs` $\ge$ `rhs`.
+  * `GT`: `lhs` $\gt$ `rhs`.
+  * `LE`: `lhs` $\le$ `rhs`.
+  * `LT`: `lhs` $\lt$ `rhs`.
+
+The values of `compare_type` have the following semantics:
+  * `FLOAT`: If element-type is floating-type, ordered floating-point comparison
+             of `lhs` and `rhs` is performed for all values of
+             `comparison_direction` except for `NE` for which unordered
+             comparison is performed. If element-type is complex-type,
+             lexicographic comparison of (real, imag) pairs is performed.
+  * `TOTALORDER`: `totalOrder` floating-point comparison of `lhs` and `rhs` as
+                  specified in IEEE-754.
+  * `SIGNED`: Signed comparison of `lhs` and `rhs`.
+  * `UNSIGNED`: Unsigned comparison of `lhs` and `rhs`.
+
+### Inputs
+
+| Name                   | Type                                                    |
+|------------------------|---------------------------------------------------------|
+| `lhs`                  | tensor of any supported type                            |
+| `rhs`                  | tensor of any supported type                            |
+| `comparison_direction` | enum of `EQ`, `NE`, `GE`, `GT`, `LE`, and `LT`          |
+| `compare_type`         | enum of `FLOAT`, `TOTALORDER`, `SIGNED`, and `UNSIGNED` |
+
+### Outputs
+
+| Name     | Type                   |
+|----------|------------------------|
+| `result` | tensor of boolean type |
+
+### Constraints
+
+  * (C1) `lhs` and `rhs` have the same element type.
+  * (C2) `lhs`, `rhs`, and `result` have the same shape.
+  * (C3) Given `E` is the `lhs` element type, the following are legal values of
+         `compare_type` and `comparison_direction`:
+    * If `E` is complex type, `compare_type` = `FLOAT` and
```

Comment:
Why can't `compare_type` be `TOTALORDER` for complex tensors?

Reply:
I am not sure of the reason, but it cant 😅
The default `FLOAT` is only allowed for complex types

Reply:
You're right! I missed [this part](https://github.com/tensorflow/tensorflow/blob/82e83ea97011c58945c8aebab969800e0d4aebf1/tensorflow/compiler/xla/service/hlo_verifier.cc#L2418-L2442) of hlo_verifier.cc. Thank you for finding out and documenting this.